edge-modules/ModuleRestarter/src/Program.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<Program>();
+        static readonly string ServiceClientConnectionString = Settings.Current.ServiceClientConnectionString;
+        static readonly string DeviceId = Settings.Current.DeviceId;
+        static readonly string DesiredModulesToRestartCSV = Settings.Current.DesiredModulesToRestartCSV;
+        static readonly int RandomRestartIntervalInMins = Settings.Current.RandomRestartIntervalInMins;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Log.LogInformation($"Starting module restarter with the following settings:\r\n{Settings.Current}");
+

```

Comment:
validate ServiceClientConnectionString is not empty?

Reply:
I added precondition checks to all the environment parameters